src/gui/tray/usermodel.cpp

```diff
@@ -502,10 +505,14 @@ void User::slotAddErrorToGui(const QString &folderAlias, SyncFileItem::Status st
         activity._accName = folderInstance->accountState()->account()->displayName();
         activity._folder = folderAlias;
 
+        // Error notifications don't have ids by themselves so we will create one for it
+        _lastErrorId -= 1;
```

Comment:
@claucambra so, it seems, even in case we'll have repeated `_subject`, `errorMessage, `folderAlias`, etc. we'll always be displaying that same notification (from the user's point of view), with a different id, given that, we don't take those properties into account as you do with file activities using `qHash()`, but, just decrementing the `_lastErrorId`.

Also, won't we have a reverse overflow at some point?

What comes to my mind is, you could use `qHash()` but then, just negate the result, so you won't need to have to decrement the `_lastErrorId` and won't have to store it. This would make the `activity._id` really depend on the `_subject`, `errorMessage`, `folderAlias`, etc.

Reply:
Thanks, now using a negative qHash